src/Microsoft.AspNet.Mvc.Core/ModelBindingHelper.cs

```diff
@@ -530,5 +524,189 @@ public static object ConvertValuesToCollectionType<T>(Type modelType, IList<T> v
+        /// <param name="type">The <see cref="Type"/> for conversion.</param>
+        /// <param name="culture">The <see cref="CultureInfo"/> for conversion.</param>
+        /// <returns>
+        /// The converted value or <c>null</c> if the value could not be converted.
+        /// </returns>
+        public static object ConvertTo(object value, [NotNull] Type type, CultureInfo culture)
+        {
+            if (value == null)
+            {
+                // For value types, treat null values as though they were the default value for the type.
+                return type.GetTypeInfo().IsValueType ? Activator.CreateInstance(type) : null;
+            }
+
+            if (value.GetType().IsAssignableFrom(type))
+            {
+                return value;
+            }
+
+            var cultureToUse = culture ?? CultureInfo.InvariantCulture;
+            return UnwrapPossibleArrayType(value, type, cultureToUse);
+        }
+
+        private static object UnwrapPossibleArrayType(object value, Type destinationType, CultureInfo culture)
+        {
+            // array conversion results in four cases, as below
+            var valueAsArray = value as Array;
+            if (destinationType.IsArray)
+            {
+                var destinationElementType = destinationType.GetElementType();
+                if (valueAsArray != null)
+                {
+                    // case 1: both destination + source type are arrays, so convert each element
+                    var converted = (IList)Array.CreateInstance(destinationElementType, valueAsArray.Length);
+                    for (var i = 0; i < valueAsArray.Length; i++)
+                    {
+                        converted[i] = ConvertSimpleType(valueAsArray.GetValue(i), destinationElementType, culture);
+                    }
+                    return converted;
+                }
+                else
+                {
+                    // case 2: destination type is array but source is single element, so wrap element in
+                    // array + convert
+                    var element = ConvertSimpleType(value, destinationElementType, culture);
+                    var converted = (IList)Array.CreateInstance(destinationElementType, 1);
+                    converted[0] = element;
+                    return converted;
+                }
+            }
+            else if (valueAsArray != null)
+            {
+                // case 3: destination type is single element but source is array, so extract first element + convert
+                if (valueAsArray.Length > 0)
+                {
+                    value = valueAsArray.GetValue(0);
+                    return ConvertSimpleType(value, destinationType, culture);
+                }
+                else
+                {
+                    // case 3(a): source is empty array, so can't perform conversion
+                    return null;
```

Comment:
If doing the `Activator.CreateInstance(type)` thing above, should do it here too.